### ISSUE TITLE
No WiFi available after finishing hotspot mode

### ISSUE THREAD

**Issue by bastos77** [ubports/ubports-touch#109]
FP2 after OTA-1 r1 update coming from legacy r41 (Version 41) this morning.
I have WiFi available before activating hotspot. But after finishing hotspot mode, there is no WiFi or SSID available anymore. After restart WiFi connection works again normally.
I have repeated this bug and it is the same.
Yesterday in legacy r41 this bug was not there. I have been able to switch between WiFi and hotspot bach to WiFi without problems.

**Comment by wagafo** [comment]
I'm seeing also this in the Nexus 5. This is what  I'm running:
channel: ubports-touch/15.04/rc
last update: 1970-01-03 18:02:07
version version: 3

**Comment by bastos77** [comment]
Today I worked with hotspot and everything went right  I am again on OTA-1 (15.04 (r1)) So bug is away for the time being :)

**Comment by wagafo** [comment]
I'm still seeing this issue in the 15.04/devel channel, revision 40.

**Comment by zubozrout** [comment]
I can confirm on MX4 as well (I am on devel). Going to airplane mode and back won't help either but reboot fortunately does.

**Comment by stekalb** [comment]
also on opo
channel: 15.04/stable and
channel: 15.04/devel (r281)

**Comment by wagafo** [comment]
I confirm on OPO, 15.04, stable and development

**Comment by cibersheep** [comment]
Same behaviour on Hammerhead Xenial 2018-W31

**Comment by userj** [comment]
Still in hammerhead 16.04/devel

**Comment by myii** [comment]
Persists in hammerhead 16.04/devel, version 796 (2019-01-15).

**Comment by hummlbach** [comment]
Cannot reproduce this on turbo@ota7

**Comment by kugiigi** [comment]
So far I can't reproduce this on my bq E5 (OTA-7).
Hotspot seems to work a lot better on it than on my Nexus 5.

**Comment by myii** [comment]
In a conversation with @kugiigi, the difference between our situations is that I had this problem with `Cellular data` on while he hadn't used that.  That suggests this issue has something to do with the bridge made between the `Wi-Fi` and `Cellular data` connections.  So to be more specific about triggering this problem:

1. Ensure both `Wi-Fi` and `Cellular data` are working.
1. Switch on the hotspot and connect from another device, using the (`Cellular data`-based) internet.
1. Switch off the hotspot.
1. `Wi-Fi` connection is no longer working properly.

@hummlbach Is this the same situation for you as well?

**Comment by zubozrout** [comment]
@myii Even though I wasn't asked I can confirm the exact same behaviour when working with a hotspot. I only use is sometimes if I need to for example connect with my laptop somewhere on the go and that's exactly what I do and when it fails.

**Comment by myii** [comment]
@zubozrout Thanks for your feedback.  I was only pinging the other users who couldn't reproduce this issue, to find out whether `Cellular data` was enabled or not when they tested.

**Comment by hummlbach** [comment]
@myii no, hotspot is really working flawlessly on turbo here. I'm using it almost everyday on the train and I think I don't have any problems connecting to and using wifis afterwards. And of course I executed your steps just before sending this message (over my wlan). :)

**Comment by myii** [comment]
@hummlbach Great, at least it isn't affecting all devices.  Thanks for your response.

**Comment by SaltyCybernaut** [comment]
I also have a Nexus 5 (hammerhead) on 16.04(OTA-7) with the identical issue. Only a reboot resolves the issue.

**Comment by luksus42** [comment]
This issue still persists... anyway I found a workaround, which may lead to a solution:
After turning hotspot off, restart unity8 by using Ubuntu Touch Tweak Tool.
Effect:
1. The hotspot indicator will appear (if it didn't already).
2. The hotspot switch is enabled again.
3. Now you can turn off hotspot again and everything works a exspected.

I think the unity8 restart will also affect thi issues #281 and #1056.
